pyannote/audio/embedding/approaches/contrastive_loss.py

```diff
@@ -28,85 +28,68 @@
+        are averaged before comparison. The intuition is that it might help
+        learn embeddings meant to be averaged/summed.
+    per_label : `int`, optional
+        Number of sequences per speaker in each batch. Defaults to 1.
+    per_fold : `int`, optional
+        Number of different speakers per batch. Defaults to 32.
+    per_epoch : `float`, optional
+        Force total audio duration per epoch, in days.
+        Defaults to total duration of protocol subset.
+    label_min_duration : `float`, optional
+        Remove speakers with less than that many seconds of speech.
+        Defaults to 0 (i.e. keep them all).
     metric : {'euclidean', 'cosine', 'angular'}, optional
         Defaults to 'cosine'.
     margin: float, optional
         Margin multiplicative factor. Defaults to 0.2.
-    size_average : `bool`, optional
-        Divide total loss by batch size
-    per_label : int, optional
-        Number of sequences per speaker in each batch. Defaults to 3.
-    label_min_duration : float, optional
-        Remove speakers with less than `label_min_duration` seconds of speech.
-        Defaults to 0 (i.e. keep them all).
-    per_fold : int, optional
-        If provided, sample triplets from groups of `per_fold` speakers at a
-        time. Defaults to sample triplets from the whole speaker set.
-    per_epoch : float, optional
-        Force total audio duration per epoch, in days.
-        Defaults to total duration of protocol subset.
+
+    Reference
+    ---------
+    TODO
     """
 
-    def __init__(self, duration=1.0,
-                       metric='cosine',
-                       margin=0.2,
-                       size_average=True,
-                       per_label=3,
-                       per_fold=None,
+    def __init__(self, duration: float = 1.0,
+                       per_turn: int = 1,
+                       per_label: int = 1,
+                       per_fold: int = 32,
                        per_epoch: float = None,
-                       label_min_duration=0.):
-
-        super().__init__()
+                       label_min_duration: float = 0.,
+                       # FIXME create a Literal type for metric
+                       # FIXME maybe in pyannote.core.utils.distance
+                       metric: str = 'cosine',
+                       # FIXME homogeneize the meaning of margin parameter
+                       # FIXME it has a different meaning in ArcFace, right?
+                       margin: float = 0.2):
+
+        super().__init__(duration=duration,
+                         per_turn=per_turn,
+                         per_label=per_label,
+                         per_fold=per_fold,
+                         per_epoch=per_epoch,
+                         label_min_duration=label_min_duration)
 
         self.metric = metric
         self.margin = margin
-
+        # FIXME see above
         self.margin_ = self.margin * self.max_distance
```

Comment:
as above, maybe we should avoid doing this kind of transformations to the margin. What about raising an exception if margin > max_distance ?